examples/wave-eager-mpi.py

```diff
@@ -0,0 +1,139 @@
+__copyright__ = "Copyright (C) 2020 Andreas Kloeckner"
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+
+import numpy as np
+import numpy.linalg as la  # noqa
+import pyopencl as cl
+
+from pytools.obj_array import flat_obj_array
+
+from meshmode.array_context import PyOpenCLArrayContext
+from meshmode.dof_array import thaw
+
+from meshmode.mesh import BTAG_ALL, BTAG_NONE  # noqa
+
+from grudge.eager import EagerDGDiscretization
+from grudge.shortcuts import make_visualizer
+from mirgecom.integrators import rk4_step
+from mirgecom.wave import wave_operator
+from mpi4py import MPI
```

Comment:
I would suggest we delete the non-MPI version then.

Reply:
Imho, we could keep both. MPI/mpi4py has some additional challenges compared to serial mirgecom so that the non-MPI variant could be useful.

Reply:
Maintaining code is not free. The more code we produce, the more there is to maintain.

Reply:
I agree with both of you.  Not free to maintain, but I would argue that it is worth it.  It is instructive to compare the non-mpi version to the mpi version.  If nothing else, this side-by-side comparison is some testament to the advantages of the general approach.

Additionally, it allows someone who does not have MPI (god help them) still run some examples.

Reply:
:shrug: OK, let's see how it plays out. :)